src/Bindings/ManualBindings.cpp

```diff
@@ -3958,22 +3958,22 @@ static int tolua_cCompositeChat_UnderlineUrls(lua_State * tolua_S)
 
 static int tolua_cEntity_IsSubmerged(lua_State * tolua_S)
 {
+	// Check the params:
 	cLuaState L(tolua_S);
-
-	// Get the params:
-	cEntity * self = reinterpret_cast<cEntity *>(tolua_tousertype(tolua_S, 1, nullptr));
-	if (self == nullptr)
+	if (!L.CheckParamSelf("cEntity"))
 	{
-		LOGWARNING("%s: invalid self (%p)", __FUNCTION__, static_cast<void *>(self));
```

Comment:
Is removing this warning OK or do I need to replace it with a new one not depending on self?

Reply:
`CheckParamSelf` give a useful error message on its own.

Reply:
Great. Everything else is fixed per your comments then. The build failed on AppVeyor due to a merge conflict, and travis didn't get triggered for some reason but it builds on my machine.

Reply:
Can you fix the merge conflict and hopefully travis will work next time.

Reply:
Done.